src/app/dataset-block/metadata-block/metadata-block.component.ts

```diff
@@ -54,14 +56,31 @@ export class MetadataBlockComponent extends BaseProcessingComponent implements O
             this.loadMetadataBlock(),
             this.loadHistory(),
             this.appDatasetSubsService.onDatasetHistoryChanges.subscribe((result: DatasetHistoryUpdate) => {
-                this.datasetHistoryUpdate = result;
+                this.mapHistory.set(result.pageInfo.currentPage, result);
+                if (result.pageInfo.hasNextPage && result.pageInfo.totalPages !== this.mapHistory.size) {
+                    this.loadHistory(result.pageInfo.currentPage + 1);
+                }
+                const firstPageHistory = this.mapHistory.get(0);
+                if (firstPageHistory) {
+                    this.datasetHistoryUpdate = firstPageHistory;
+                }
```

Comment:
Try implementing a _lazy load_ algorithm. It should request and cache the next page only if really needed, if the consumer keeps asking for more data. 

Imagine you have a history of a few hundred pages, and the answer to the filter, like 10 events of certain type, is satisfied by the time you processed page 3 (let's say you've found enough nodes to show a page of filtered results), It would be unnecessary to continue loading history. But maybe user requests another 10 events of this type, so you would continue loading to page 8 to satisfy it. Loading hundreds of pages still does not look reasonable.